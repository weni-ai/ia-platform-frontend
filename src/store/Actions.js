import nexusaiAPI from '@/api/nexusaiAPI.js';
import { defineStore } from 'pinia';
import { computed, reactive } from 'vue';
import globalStore from '.';

export const useActionsStore = defineStore('actions', () => {
  const connectProjectUuid = computed(
    () => globalStore.state.Auth.connectProjectUuid,
  );

  const actions = reactive({
    status: null,
    data: [],
  });

  async function loadActions() {
    if (actions.status !== null) {
      return;
    }

    try {
      actions.status = 'loading';

      const data = await nexusaiAPI.router.actions.list({
        projectUuid: connectProjectUuid.value,
      });

      actions.data = data;

      actions.status = 'complete';
    } catch (error) {
      actions.status = 'error';
    }
  }

  async function addAction({ name, prompt, flowUuid, type }) {
    const action = await nexusaiAPI.router.actions.create({
      projectUuid: connectProjectUuid.value,
      name,
      prompt,
      flowUuid,
      type,
    });

    actions.data.push(action);

    return action;
  }

  async function editAction({ uuid, name, prompt }) {
    const editedAction = await nexusaiAPI.router.actions.edit({
      projectUuid: connectProjectUuid.value,
      actionUuid: uuid,
      name,
      prompt,
    });

    const item = actions.data.find(
      (action) => action.uuid === editedAction.uuid,
    );

    item.name = editedAction.name;
    item.prompt = editedAction.prompt;

    return editedAction;
  }

  async function removeAction({ uuid }) {
    await nexusaiAPI.router.actions.delete({
      projectUuid: connectProjectUuid.value,
      actionUuid: uuid,
    });

    actions.data = actions.data.filter((action) => action.uuid !== uuid);
  }

  const types = reactive({
    status: null,
    data: [],
  });

  const typesAvailable = computed(() => {
    return types.data.filter(
      (type) =>
        !actions.data.some(
          (action) =>
            action.name === type.name && action.prompt === type.prompt,
        ),
    );
  });

  async function loadTypes() {
    if (types.status !== null) {
      return;
    }

    try {
      types.status = 'loading';

      const data = await nexusaiAPI.router.actions.types.list({
        projectUuid: connectProjectUuid.value,
      });

      types.data = data;

      types.status = 'complete';
    } catch {
      types.status = 'error';
    }
  }

  return {
    actions,
    load: loadActions,
    add: addAction,
    edit: editAction,
    remove: removeAction,

    types,
    typesAvailable,
    loadTypes,
  };
});
