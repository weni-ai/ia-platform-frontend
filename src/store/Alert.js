import { defineStore } from 'pinia';
import { reactive, ref } from 'vue';

export const useAlertStore = defineStore('alert', () => {
  const id = ref(0);
  const alert = reactive({});

  function addAlert({ text, type }) {
    id.value += 1;

    alert.text = text;
    alert.type = type;
  }

  function closeAlert() {
    alert.text = '';
    alert.type = '';
  }

  return {
    id,
    data: alert,

    add: addAlert,
    close: closeAlert,
  };
});
