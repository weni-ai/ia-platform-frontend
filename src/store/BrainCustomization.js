import nexusaiAPI from '@/api/nexusaiAPI';
import { defineStore } from 'pinia';
import { computed, reactive, ref, watch } from 'vue';
import globalStore from '.';
import { cloneDeep, differenceBy, get } from 'lodash';

export const useBrainCustomizationStore = defineStore(
  'brainCustomization',
  () => {
    const connectProjectUuid = computed(
      () => globalStore.state.Auth.connectProjectUuid,
    );

    const status = ref(null);
    const isSaving = ref(false);

    const name = reactive({
      current: '',
      old: '',
    });

    const role = reactive({
      current: '',
      old: '',
    });

    const personality = reactive({
      current: '',
      old: '',
    });

    const goal = reactive({
      current: '',
      old: '',
    });

    const instructions = reactive({
      current: '',
      old: '',
    });

    const errorRequiredFields = reactive({
      name: false,
      role: false,
      goal: false,
    });

    const hasChanged = computed(() => {
      return (
        [name, role, personality, goal].some(
          ({ current, old }) => current !== old,
        ) ||
        !!differenceBy(instructions.current, instructions.old, 'instruction')
          .length
      );
    });

    const isSaveButtonDisabled = computed(() => {
      const hasErrorRequiredFields =
        Object.values(errorRequiredFields).includes(true);

      return !hasChanged.value || hasErrorRequiredFields;
    });

    function setInitialValues(data) {
      name.old = name.current = data.agent?.['name'] || '';
      role.old = role.current = data.agent?.['role'] || '';
      personality.old = personality.current = data.agent?.['personality'] || '';
      goal.old = goal.current = data.agent?.['goal'] || '';

      instructions.current = data.instructions || [];

      if (instructions.current.length === 0) {
        instructions.current.push({
          instruction: '',
        });
      }

      instructions.old = cloneDeep(instructions.current);
    }

    async function load() {
      if (status.value !== null) {
        return;
      }

      try {
        status.value = 'loading';

        const { data } = await nexusaiAPI.router.customization.read({
          projectUuid: connectProjectUuid.value,
        });

        setInitialValues(data);

        status.value = 'complete';
      } catch {
        status.value = 'error';
      }
    }

    function validate() {
      const fields = [
        {
          id: 'name',
          field: name,
        },
        {
          id: 'role',
          field: role,
        },
        {
          id: 'goal',
          field: goal,
        },
      ];

      const unfilledFields = fields.filter(
        ({ field }) => !field.current.trim(),
      );

      unfilledFields.forEach(({ id, field }) => {
        errorRequiredFields[id] = true;

        watch(
          () => field.current,
          () => {
            errorRequiredFields[id] = false;
          },
          { once: true },
        );
      });

      if (unfilledFields.length) {
        throw {
          tab: 'personalization',
        };
      }
    }

    async function save() {
      try {
        validate();

        isSaving.value = true;

        const payload = {
          agent: {
            name: name.current,
            role: role.current,
            personality: personality.current,
            goal: goal.current,
          },
          instructions: instructions.current.filter(
            ({ instruction }) => instruction,
          ),
        };

        const { data } = await nexusaiAPI.router.customization.edit({
          projectUuid: connectProjectUuid.value,
          data: payload,
        });

        setInitialValues(data);
      } catch (error) {
        const tabWithError =
          get(error, 'tab') ||
          {
            'brain-customization-edit': 'personalization',
          }[get(error, 'config.routerName')];

        if (tabWithError) {
          globalStore.state.Brain.tabsWithError = [tabWithError];
        }
      } finally {
        isSaving.value = false;
      }
    }

    async function removeInstruction(instructionIndex) {
      const { id: removeId } = instructions.current[instructionIndex];

      if (removeId) {
        await nexusaiAPI.router.customization.delete({
          projectUuid: connectProjectUuid.value,
          id: removeId,
        });
      }

      instructions.current.splice(instructionIndex, 1);
      instructions.old.splice(instructionIndex, 1);
    }

    function addEmptyInstruction() {
      instructions.current.push({
        instruction: '',
      });

      instructions.old.push({
        instruction: '',
      });
    }

    return {
      status,
      isSaving,
      name,
      role,
      personality,
      goal,
      instructions,
      errorRequiredFields,
      hasChanged,
      isSaveButtonDisabled,

      load,
      validate,
      save,

      removeInstruction,
      addEmptyInstruction,
    };
  },
);
