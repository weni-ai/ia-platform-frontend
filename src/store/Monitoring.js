import { defineStore } from 'pinia';
import { computed, reactive } from 'vue';
import { useRoute } from 'vue-router';

import nexusaiAPI from '@/api/nexusaiAPI.js';
import globalStore from '.';
import i18n from '@/utils/plugins/i18n';

export const useMonitoringStore = defineStore('monitoring', () => {
  const connectProjectUuid = computed(
    () => globalStore.state.Auth.connectProjectUuid,
  );

  const route = useRoute();

  const messages = reactive({
    status: null,
    data: [],
    count: 0,
    newMessages: [],
    performance: {
      status: null,
      action: 0,
      success: 0,
      failed: 0,
    },
    inspectedAnswer: {},
  });

  async function loadMessages({ page, pageInterval, tag, text }) {
    const { started_day, ended_day } = route.query;
    const currentNewMessages = [...messages.newMessages];

    try {
      messages.status = 'loading';
      messages.newMessages = [];

      const { data, count } = await nexusaiAPI.router.monitoring.messages.list({
        projectUuid: connectProjectUuid.value,
        page,
        pageInterval,
        tag,
        text,
        started_day,
        ended_day,
      });

      messages.data = data;
      messages.count = count;
      messages.status = 'complete';
    } catch (error) {
      messages.status = 'error';
      messages.newMessages = currentNewMessages;
    }
  }

  async function loadMessageContext({ id }) {
    messages.inspectedAnswer.context = {};
    const setStatus = (status) =>
      (messages.inspectedAnswer.context.status = status);

    try {
      setStatus('loading');

      const response =
        await nexusaiAPI.router.monitoring.messages.getMessageContext({
          projectUuid: connectProjectUuid.value,
          id: id,
        });

      messages.inspectedAnswer.context = response?.map(
        ({ id, llm_response, tag }) => ({
          id,
          llm: { response: llm_response, status: tag },
        }),
      );

      setStatus('complete');
    } catch (error) {
      setStatus('error');

      globalStore.state.alert = {
        type: 'error',
        text: i18n.global.t(
          'router.monitoring.error_loading_previous_messages',
        ),
      };
    }
  }

  async function loadMessageDetails({ id }) {
    try {
      messages.inspectedAnswer.status = 'loading';

      const response = await nexusaiAPI.router.monitoring.messages.detail({
        projectUuid: connectProjectUuid.value,
        id,
      });

      const {
        uuid,
        text,
        status,
        actions_started,
        actions_uuid,
        actions_type,
        llm_response,
        is_approved,
        contact_urn,
        groundedness,
      } = response;

      const llmStatusMap = {
        s: 'success',
        f: 'failed',
      };

      messages.inspectedAnswer = {
        id,
        uuid,
        text,
        action: actions_started
          ? {
              name: actions_type,
              uuid: actions_uuid,
            }
          : null,
        llm: {
          response: llm_response,
          status: actions_started
            ? 'action'
            : llmStatusMap[status.toLowerCase()],
        },
        contact_urn,
        is_approved,
        groundedness: groundedness?.map((item) => ({
          ...item,
          score: Number(item.score) * 10,
        })), // To turn into a hundred

        status: 'complete',
      };
    } catch (error) {
      messages.inspectedAnswer.status = 'error';
      globalStore.state.alert = {
        type: 'error',
        text: i18n.global.t('router.monitoring.error_loading_messages'),
      };
    }
  }

  async function loadMessagesPerformance() {
    const { started_day, ended_day } = route.query;
    try {
      messages.performance.status = 'loading';

      const response = await nexusaiAPI.router.monitoring.messages.performance({
        projectUuid: connectProjectUuid.value,
        started_day,
        ended_day,
      });

      messages.performance = {
        status: 'complete',
        action: response?.action_percentage || 0,
        success: response?.succeed_percentage || 0,
        failed: response?.failed_percentage || 0,
      };
    } catch (error) {
      messages.performance.status = 'error';
    }
  }

  async function rateAnswer({ is_approved }) {
    try {
      const response = await nexusaiAPI.router.monitoring.messages.rateAnswer({
        projectUuid: connectProjectUuid.value,
        id: messages.inspectedAnswer.id,
        is_approved,
      });

      messages.inspectedAnswer.is_approved = response.is_approved;
    } catch (error) {
      throw new Error(error);
    }
  }

  function createNewMessage({ message }) {
    messages.newMessages.push(message);
  }

  return {
    messages,
    loadMessages,
    loadMessageContext,
    loadMessageDetails,
    loadMessagesPerformance,
    rateAnswer,
    createNewMessage,
  };
});
