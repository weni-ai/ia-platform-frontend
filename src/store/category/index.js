import actions from './actions';

export default {
  actions,

  state: () => ({
    allCategories: null,
  }),
};
