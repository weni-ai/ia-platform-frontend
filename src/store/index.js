import { createStore } from 'vuex';

import Auth from './auth';
import User from './user';
import Repository from './repository';
import Category from './category';
import CachedFetch from './cached-fetch';
import News from './news';
import Integration from './integration';
import Brain from './brain';

export const store = createStore({
  state() {
    return {
      obstructiveError: null,

      alert: null,

      modalWarn: null,

      router: {
        intelligenceUuid: null,
        contentBaseUuid: null,
      },
    };
  },

  modules: {
    Auth,
    User,
    Repository,
    Category,
    CachedFetch,
    News,
    Integration,
    Brain,
  },
});

store.dispatch('retriveAuthToken');

export default store;
