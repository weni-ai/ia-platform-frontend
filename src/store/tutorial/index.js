import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';

export default {
  state: () => state,
  getters,
  actions,
  mutations,
};
