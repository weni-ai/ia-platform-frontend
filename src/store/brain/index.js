import { differenceBy, cloneDeep, pick, get } from 'lodash';
import { WENIGPT_OPTIONS } from '../../utils';
import { models } from './models.js';
import nexusaiAPI from '../../api/nexusaiAPI.js';
import Vue from 'vue';
import i18n from '../../utils/plugins/i18n';
import store from '../index.js';

function watchOnce(...args) {
  const unwatch = store.watch(args[0], () => {
    args[1]();
    unwatch();
  });
}

export default {
  state: {
    tabsWithError: null,
    isSavingChanges: false,
    customizationStatus: 'waitingToLoad',
    tuningsStatus: 'waitingToLoad',

    contentText: {
      uuid: null,
      current: '',
      old: '',
    },

    agent: {
      name: {
        current: '',
        old: '',
      },
      role: {
        current: '',
        old: '',
      },
      goal: {
        current: '',
        old: '',
      },
      personality: {
        current: '',
        old: '',
      },
    },

    instructions: {
      current: [],
      old: [],
    },

    tunings: {},
    tuningsOld: {},

    customizationErrorRequiredFields: {
      name: false,
      role: false,
      goal: false,
    },
  },

  getters: {
    hasBrainCustomizationChanged(state) {
      return (
        Object.keys(state.agent).some((key) => {
          const { current, old } = state.agent[key];
          return current !== old;
        }) ||
        !!differenceBy(
          state.instructions.current,
          state.instructions.old,
          'instruction',
        ).length
      );
    },

    hasBrainTuningsChanged(_state, getters) {
      return getters.brainTuningsFields.some(
        ({ value, previousValue }) => value !== previousValue,
      );
    },

    hasBrainContentTextChanged(state) {
      return state.contentText.current !== state.contentText.old;
    },

    isBrainSaveButtonDisabled(state, getters) {
      const hasCustomizationErrorRequiredFields = Object.values(
        state.customizationErrorRequiredFields,
      ).includes(true);

      return (
        (!getters.hasBrainCustomizationChanged &&
          !getters.hasBrainTuningsChanged &&
          !getters.hasBrainContentTextChanged) ||
        hasCustomizationErrorRequiredFields
      );
    },

    brainTuningsFields({ tunings, tuningsOld }) {
      const useValue = (field) => {
        const value = tunings[field.name] || field.default;
        const previousValue = tuningsOld[field.name] || field.default;

        return {
          ...field,
          value,
          previousValue,
        };
      };

      const model = useValue({
        type: 'radio',
        name: 'model',
        options: models.map(({ name }) => name),
        default: models[0].name,
      });

      return [model].concat([
        ...models.find(({ name }) => name === model.value).fields.map(useValue),
      ]);
    },
  },

  actions: {
    async loadBrainCustomization({ commit, state, rootState }) {
      if (state.customizationStatus !== 'waitingToLoad') {
        return;
      }

      try {
        state.customizationStatus = 'loading';

        const { data } = await nexusaiAPI.router.customization.read({
          projectUuid: rootState.Auth.connectProjectUuid,
        });

        let currentData = data;

        if (currentData.agent === null) {
          currentData.agent = {
            name: '',
            role: '',
            goal: '',
            personality: '',
          };
        }

        commit('setBrainCustomizationInitialValues', currentData);
      } catch (error) {
        rootState.alert = {
          type: 'error',
          text: i18n.t('customization.invalid_get_data'),
        };
      } finally {
        state.customizationStatus = 'loaded';
      }
    },

    async loadBrainTunings({ commit, state, rootState }) {
      if (state.tuningsStatus !== 'waitingToLoad') {
        return;
      }

      try {
        state.tuningsStatus = 'loading';

        const { data } = await nexusaiAPI.router.tunings.read({
          projectUuid: rootState.Auth.connectProjectUuid,
        });

        commit('setBrainTuningsInitialValues', data);
      } finally {
        state.tuningsStatus = 'loaded';
      }
    },

    validateBrainCustomization({ state }) {
      const unfilledFields = ['name', 'role', 'goal'].filter(
        (property) => !state.agent[property].current,
      );

      unfilledFields.forEach((property) => {
        state.customizationErrorRequiredFields[property] = true;

        watchOnce(
          ({ Brain }) => Brain.agent[property].current,
          () => {
            state.customizationErrorRequiredFields[property] = false;
          },
        );
      });

      if (unfilledFields.length) {
        throw {
          tab: 'personalization',
        };
      }
    },

    async saveBrainCustomization({ state, commit, rootState }) {
      const agent = Object.keys(state.agent).reduce(
        (obj, key) => ({
          ...obj,
          [key]: state.agent[key].current,
        }),
        {},
      );

      const currentValue = {
        agent,
        instructions: state.instructions.current.filter((e) => e.instruction),
      };

      const { data } = await nexusaiAPI.router.customization.edit({
        projectUuid: rootState.Auth.connectProjectUuid,
        data: currentValue,
      });

      commit('setBrainCustomizationInitialValues', data);
    },

    async saveBrainContentText({ state, commit, rootState }) {
      if (state.contentText.uuid) {
        const { data: contentBaseTextData } =
          await nexusaiAPI.updateIntelligenceContentBaseText({
            contentBaseUuid: rootState.router.contentBaseUuid,
            contentBaseTextUuid: state.contentText.uuid,
            text: state.contentText.current,
            hideGenericErrorAlert: true,
          });

        commit('setBrainContentTextInitialValues', contentBaseTextData);
      } else {
        const { data: contentBaseTextData } =
          await nexusaiAPI.createIntelligenceContentBaseText({
            contentBaseUuid: rootState.router.contentBaseUuid,
            text: state.contentText.current,
            hideGenericErrorAlert: true,
          });

        commit('setBrainContentTextInitialValues', contentBaseTextData);
      }
    },

    async saveBrainTunings({ getters, commit, rootState }) {
      const handleFieldName = (name) =>
        name === 'version-gpt' ? 'version' : name;
      const handleFieldValue = (name, value) =>
        name === 'version'
          ? WENIGPT_OPTIONS.find((e) => e.name === value).model
          : value;

      const { data } = await nexusaiAPI.router.tunings.edit({
        projectUuid: rootState.Auth.connectProjectUuid,
        values: getters.brainTuningsFields.reduce(
          (values, field) => ({
            ...values,
            [handleFieldName(field.name)]: handleFieldValue(
              field.name,
              field.value,
            ),
          }),
          {},
        ),
      });

      const fieldsChangeds = getters.brainTuningsFields
        .filter(({ value, previousValue }) => value !== previousValue)
        .map((field) => pick(field, ['name', 'value', 'type']));

      commit('setBrainTuningsInitialValues', data);

      if (window.brainPreviewAddMessage) {
        fieldsChangeds.forEach((field) => {
          const name =
            {
              model: 'router.tunings.model',
            }[field.name] || `router.tunings.fields.${field.name}`;

          window.brainPreviewAddMessage({
            type: 'change',
            name,
            value:
              field.type === 'password'
                ? field.value.replace(/./g, 'â€¢')
                : field.value,
          });
        });
      }
    },

    async saveBrainChanges({ state, getters, rootState, dispatch }) {
      try {
        getters.hasBrainCustomizationChanged &&
          (await dispatch('validateBrainCustomization'));

        state.isSavingChanges = true;

        const promises = [];

        if (getters.hasBrainCustomizationChanged) {
          promises.push(dispatch('saveBrainCustomization'));
        }

        if (getters.hasBrainTuningsChanged) {
          promises.push(dispatch('saveBrainTunings'));
        }

        if (getters.hasBrainContentTextChanged) {
          promises.push(dispatch('saveBrainContentText'));
        }

        const tabsWithError = (await Promise.allSettled(promises))
          .filter(({ status }) => status === 'rejected')
          .map(({ reason }) => get(reason, 'config.routerName'))
          .map(
            (routerName) =>
              ({
                'brain-customization-edit': 'personalization',
                'contentBase-text-create': 'content',
                'contentBase-text-edit': 'content',
                'brain-tunings-edit': 'tunings',
              }[routerName]),
          )
          .filter((tab) => tab);

        if (tabsWithError.length) {
          state.tabsWithError = tabsWithError;
        } else {
          rootState.alert = {
            type: 'success',
            text: i18n.t('router.tunings.changes_saved'),
          };
        }
      } catch (error) {
        const tabWithError = get(error, 'tab');

        if (tabWithError) {
          state.tabsWithError = [tabWithError];
        }
      } finally {
        state.isSavingChanges = false;
      }
    },
  },

  mutations: {
    setBrainCustomizationInitialValues(state, data) {
      Object.keys(data.agent).forEach((key) => {
        state.agent[key].old = state.agent[key].current = data.agent[key] || '';
      });

      state.instructions.current = data.instructions || [];

      if (state.instructions.current.length === 0) {
        state.instructions.current.push({
          instruction: '',
        });
      }

      state.instructions.old = cloneDeep(state.instructions.current);
    },

    setBrainTuningsInitialValues(state, data) {
      const handleName = (name) =>
        name === 'version' && data.model === 'ChatGPT' ? 'version-gpt' : name;
      const getValue = (name) => {
        const specialValues = ['temperature', 'top_p', 'top_k'];

        if (specialValues.includes(name)) {
          return Number(data.setup[name]);
        }

        if (name === 'version' && data.model === 'WeniGPT') {
          const option = WENIGPT_OPTIONS.find(
            (value) => value.model === data.setup[name],
          );
          return option ? option.name : '';
        }
        return data.setup[name];
      };

      Vue.set(state.tunings, 'model', data.model);
      Vue.set(state.tuningsOld, 'model', data.model);

      Object.keys(data.setup).forEach((name) => {
        Vue.set(state.tunings, handleName(name), getValue(name));
        Vue.set(state.tuningsOld, handleName(name), getValue(name));
      });
    },

    setBrainContentTextInitialValues(state, data) {
      state.contentText.uuid = data.uuid;
      state.contentText.current = state.contentText.old = data.text;
    },
  },
};
