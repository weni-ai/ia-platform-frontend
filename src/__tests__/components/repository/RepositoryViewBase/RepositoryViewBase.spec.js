import { shallowMount, createLocalVue } from '@vue/test-utils';
import RepositoryViewBase from '@/components/repository/RepositoryViewBase/RepositoryViewBase';
import Repository from '@/models/repository';
import store from '@/store';
import Router from 'vue-router';
import Buefy from 'buefy';

const localVue = createLocalVue();
localVue.use(Buefy);

localVue.use(Router);

describe('RepositoryViewBase.vue', () => {
  let repository;
  let wrapper;
  beforeEach(() => {
    repository = new Repository({
      ready_for_train: false,
      requirements_to_train: {
        en: [
          'There was no change in this bot version. No examples or translations for English have been added or removed.',
        ],
      },
      languages_ready_for_train: {
        en: false,
      },
      languages_warnings: {},
    });

    wrapper = shallowMount(RepositoryViewBase, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        repository,
      },
      store,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('loading', () => {
    beforeEach(() => {
      wrapper.setProps({
        repository: null,
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('error 404', () => {
    beforeEach(() => {
      wrapper.setProps({
        repository: null,
        errorCode: 404,
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('error -1', () => {
    beforeEach(() => {
      wrapper.setProps({
        repository: null,
        errorCode: -1,
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
