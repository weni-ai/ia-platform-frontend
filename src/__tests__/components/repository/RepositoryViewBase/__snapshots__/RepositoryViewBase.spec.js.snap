// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryViewBase.vue error -1 renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`RepositoryViewBase.vue error 404 renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`RepositoryViewBase.vue loading renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`RepositoryViewBase.vue renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
